File: py/Node.py
class Node:
    def __init__(self, ID, lat, lon):
        self.ID = ID
        self.lat = lat
        self.lon = lon 
        self.previous = None # of type Node 
        self.dist_from_start = 0
        self.marked = False 
        self.adjacents = []

    def setDistFromStart(self, d):
        self.dist_from_start = d
    
    def addAdjacentNode(self, v):
        if (v not in self.adjacents):
            self.adjacents.append(v)
    
    def setPrevious(self, p):
        self.previous = p 

    def getPrevious(self):
        return self.previous 

    def getID(self):
        return self.ID 

    def getLatitude(self):
        return self.lat 

    def getLongitude(self):
        return self.lon 

    def getDistFromStart(self):
        return self.dist_from_start

    def mark(self, m):
        self.marked = m

    def getMark(self):
        return self.marked
    
    def getAdjacentNode(self, i):
        return self.adjacents[i]

    def getNbAdjacents(self):
        return len(self.adjacents)

    def show(self):
        print("node {0}, marked = {1}".format(self.ID, self.marked))

    def __repr__(self):
        a = [node.getID() for node in self.adjacents]
        return "Node {0}, adj : {1}, marked : {2})".format(self.ID, a, self.marked)

File: version_c++/Node.cpp
//
// Created by Alexandre on 12-11-20.
//

#include "Node.h"

#define G_INT_MAXI 10000000

// ============================ Node ==========================

// Node class constructor
Node::Node(int id, float lat, float lon) :
        _id(id), _lat(lat), _lon(lon), _previous(nullptr),
        _distanceFromStart(G_INT_MAXI), _marked(false) {}

// Setters
void Node::setDistFromStart(float d) {_distanceFromStart = d;}
void Node::mark(bool m) {_marked = m;}
void Node::addAdjacentNode(Node* v) {_adjacents.push_back(v);}
void Node::setPrevious(Node* p) {_previous = p;}

// Getters
Node* Node::getPrevious() const {return _previous;}
const int Node::getId() const {return _id;}
const float Node::getLatitude() const {return _lat;}
const float Node::getLongitude() const {return _lon;}
float Node::getDistFromStart() const {return _distanceFromStart;}
bool Node::getMark() const {return _marked;}
Node* Node::getAdjacentNode(int i) const {return _adjacents.at(i);}
int Node::getNbAdjacents() const {return _adjacents.size();}

void Node::show() const {
    cout << "node " << _id << ", " << endl;
}